airflow/contrib/operators/segment_track_event_operator.py

```
@@ -0,0 +1,70 @@
+# under the License.
+
+from airflow.contrib.hooks.segment_hook import SegmentHook
+from airflow.models import BaseOperator
+from airflow.utils.decorators import apply_defaults
+
+
+class SegmentTrackEventOperator(BaseOperator):
+    """
+    Send Track Event to Segment for a specified user_id and event
+
+    :param user_id: The ID for this user in your database
+    :type user_id: string
+    :param event: The name of the event you're tracking
+    :type event: string
+    :param properties: A dictionary of properties for the event.
+    :type properties: dict
+    :param segment_conn_id: The connection ID to use when connecting to Segment.
+    :type segment_conn_id: string
+    :param segment_debug_mode: Determines whether Segment should run in debug mode.
+        Defaults to False
+    :type segment_debug_mode: boolean
+    """
+    template_fields = ('user_id', 'event', 'properties')
+    ui_color = '#ffd700'
+
+    @apply_defaults
+    def __init__(self,
+                 user_id,
+                 event,
+                 properties=None,
+                 segment_conn_id='segment_default',
+                 segment_debug_mode=False,
+                 *args,
+                 **kwargs):
+        super(SegmentTrackEventOperator, self).__init__(*args, **kwargs)
+        self.user_id = user_id
+        self.event = event
+        properties = properties or {}
+        self.properties = properties
+        self.segment_debug_mode = segment_debug_mode
+        self.segment_conn_id = segment_conn_id
+
+    def execute(self, context):
+        hook = SegmentHook(segment_conn_id=self.segment_conn_id,
+                           segment_debug_mode=self.segment_debug_mode)
+
+        self.log.info(
+            'Sending track event ({0}) for user id: {1} with properties: {2}'.
+            format(self.event, self.user_id, self.properties))
+
+        response = hook.track(self.user_id, self.event, self.properties)
```

Maybe check if the response is what we expect? If might otherwise fail silently

---

@Fokko looked into it and it turns out Segment doesn't really return responses from there, that's why they have the opt in to the `on_error` method that I use above. So I rewrote it to not log responses since they aren't meaningful

---

Ah, I see. Maybe as a future improvement of the API, it would be good to check if the hook returns a http 2xx.